src/components/buttons/CopyButton/CopyButton.tsx

```diff
@@ -62,34 +71,63 @@ export const CopyButton = (props: ICopyButtonProps) => {
 	const Tag: any = tag;
 
 	const [isCopied, setIsCopied] = useState(false);
+	const [popin, setPopin] = useState(false);
+	const [popout, setPopout] = useState(false);
+	const setPopinPopout = (popin: boolean, popout: boolean) => {
+		setPopin(popin);
+		setPopout(popout);
+	}
+
+	const isTransparentUncopiedStateBG = uncopiedStateBGStyleVariant === UncopiedStateBGStyleVariants.transparent;
+	const isGrayUncopiedStateBG = uncopiedStateBGStyleVariant === UncopiedStateBGStyleVariants.gray;
 
-	const isTransparentCopiedStateBG = copiedStateBGStyleVariant === CopiedStateBGStyleVariants.transparent;
+	const isGrayCopiedStateText = copiedStateBGStyleVariant === CopiedStateBGStyleVariants.transparentGrayText;
+	const isTransparentCopiedStateBG = copiedStateBGStyleVariant === CopiedStateBGStyleVariants.transparent || isGrayCopiedStateText;
 	const isGreenCopiedStateBG = copiedStateBGStyleVariant === CopiedStateBGStyleVariants.green;
+
 	const smallPadding = padding === PaddingAmount.small;
 	const mediumPadding = padding === PaddingAmount.medium;
 
+	const handleClick = () => {
+		copy(textToCopy);
+		if (isCopied) return;
+
+		const newTimeout = copiedTimeout + 2000;
+		const animationDuration = 475;
+
+		setPopout(true);
+		setTimeout(() => { 
+			setIsCopied(true);
+			setPopinPopout(true, false);
```

Comment:
Looks like the value for `popin` & `popout` are always opposite, so I'm wondering if we can combine this into a shared variable? 

Something like: 
```
const handleClick = () => {
		copy(textToCopy);
		if (isCopied) return;

		const newTimeout = copiedTimeout + 2000;
		const animationDuration = 500;

		setTimeout(() => {
			setIsCopied(true);
			setPopin(true);
		}, animationDuration);
		setTimeout(() => setPopin(false), newTimeout - animationDuration)
		setTimeout(() => {
			setIsCopied(false);
			setPopin(true);
		}, newTimeout);
		setTimeout(() => setPopin(false), newTimeout + animationDuration);
	}
```

Then for the styles, what do you think of getting rid of the `popout` class & keyframes altogether and instead utilizing the `alternate` animation-direction? Something like: 

```
@keyframes popin {
	0% { transform: scale(0); opacity: 0; }
	40% { transform: scale(1.05); opacity: 1; }
	70% { transform: scale(.95); }
	100% { transform: scale(1); }
}

.popin {
	animation: popin .5s;
	animation-direction: alternate;
}
```

Reply:
Mainly just trying to simplify this logic a bit

Reply:
I'm just trying to think of ways to simplify this logic a bit 😃

Reply:
Ugh yeah I hit a refactoring wall at this point - just toggling the class won't work because the animation has a single cycle that's triggered only when the class is applied. In my mind you should still be able to take the class off and immediately reapply it using a single variable, but `alternate` really only seems to apply when you have more than one cycle in the animation - like the next time you reapply the class it won't be in reverse.

I did try using just one `keyframes` declaration, defining the animation in the `popout` class to use those keyframes in reverse, but even that doesn't work. The `keyframes` seem to only want to be used by one thing at a time. So separating everything out very explicitly is the only way it likes to behave :/